tidepool_data_science_models/models/icgm_sensor_generator.py

```diff
@@ -137,25 +151,52 @@ def fit(self, true_bg_trace=None):
         self.batch_sensor_brute_search_results = batch_sensor_brute_search_results
         self.dist_params = self.batch_sensor_brute_search_results[0]
 
+        # %% add in check that optimal result passed
+        self.generate_sensors(self.batch_training_size, sensor_start_datetime=0)
+        self.icgm_traces_used_in_training = copy.deepcopy(self.icgm_traces)
+        self.icgm_traces = None
+
+        temp_df = sf.preprocess_data(
+            true_bg_trace, self.icgm_traces_used_in_training, icgm_range=[40, 400], ysi_range=[0, 900]
+        )
+
+        self.icgm_special_controls_accuracy_table = sf.calc_icgm_sc_table(temp_df, "generic")
+
+        if self.use_g6_accuracy_in_loss:
+            self.g6_loss, self.g6_table = sf.calc_dexcom_loss(temp_df, self.batch_training_size)
+
+        else:
+            self.g6_loss, self.g6_table = np.nan, np.nan
+
+        self.loss_of_best_search, self.percent_pass = sf.calc_icgm_special_controls_loss(
+            self.icgm_special_controls_accuracy_table, self.g6_loss
+        )
+
         return
 
     def generate_sensors(self, n_sensors, sensor_start_datetime, sensor_start_time_index=0):
 
         if self.dist_params is None:
             raise Exception("iCGM Sensor Generator has not been fit() to a true_bg_trace distribution.")
 
-        (
-            a,
-            b,
-            mu,
-            sigma,
-            noise_coefficient,
-            bias_drift_range_min,
-            bias_drift_range_max,
-            bias_drift_oscillations,
-        ) = self.dist_params
-
-        bias_drift_range = [bias_drift_range_min, bias_drift_range_max]
+        # (
```

Comment:
I think initially storing the dist parameters in an array like this is error-prone and would have advocated against it, but rather than totally reworking it I'd vote for keeping the specifically named variables for clarity and debugging purposes.